src/lib/es2015.proxy.d.ts

```diff
@@ -1,18 +1,17 @@
 interface ProxyHandler<T extends object> {
     getPrototypeOf? (target: T): object | null;
-    setPrototypeOf? (target: T, v: any): boolean;
+    setPrototypeOf? (target: T, v: object | null): boolean;
     isExtensible? (target: T): boolean;
     preventExtensions? (target: T): boolean;
-    getOwnPropertyDescriptor? (target: T, p: PropertyKey): PropertyDescriptor | undefined;
-    has? (target: T, p: PropertyKey): boolean;
-    get? (target: T, p: PropertyKey, receiver: any): any;
-    set? (target: T, p: PropertyKey, value: any, receiver: any): boolean;
-    deleteProperty? (target: T, p: PropertyKey): boolean;
-    defineProperty? (target: T, p: PropertyKey, attributes: PropertyDescriptor): boolean;
-    enumerate? (target: T): PropertyKey[];
-    ownKeys? (target: T): PropertyKey[];
-    apply? (target: T, thisArg: any, argArray?: any): any;
-    construct? (target: T, argArray: any, newTarget?: any): object;
+    getOwnPropertyDescriptor? (target: T, p: string | symbol): PropertyDescriptor | undefined;
+    has? (target: T, p: string | symbol): boolean;
+    get? (target: T, p: string | symbol, receiver: any): any;
+    set? (target: T, p: string | symbol, value: any, receiver: any): boolean;
+    deleteProperty? (target: T, p: string | symbol): boolean;
+    defineProperty? (target: T, p: string | symbol, attributes: PropertyDescriptor): boolean;
+    ownKeys? (target: T): ArrayLike<string | symbol>;
+    apply? (target: T & ((...args: any[]) => any), thisArg: any, argArray: any[]): any;
+    construct? (target: T & (new (...args: any[]) => object), argArray: any[], newTarget?: any): object;

```

Comment:
These&nbsp;will&nbsp;only ever&nbsp;get&nbsp;called if&nbsp;the&nbsp;wrapped&nbsp;object is&nbsp;callable or&nbsp;constructable, respectively.